src/pages/research.astro

```diff
@@ -0,0 +1,1131 @@
+                            <span class="ml-2 text-sm font-bold text-gray-500 move-button">
+                              DOI
+                            </span>
+                            <svg
+                              xmlns="http://www.w3.org/2000/svg"
+                              class="h-5 w-5"
+                              fill="none"
+                              viewBox="0 0 24 24"
+                              stroke="currentColor"
+                              stroke-width="2"
+                            >
+                              {/* source https://heroicons.com v1.0.6, MIT licensed */}
+                              <path
+                                stroke-linecap="round"
+                                stroke-linejoin="round"
+                                d="M10 6H6a2 2 0 00-2 2v10a2 2 0 002 2h10a2 2 0 002-2v-4M14 4h6m0 0v6m0-6L10 14"
+                              />
+                            </svg>
+                          </a>
+                        )}
+                        {paper.abstract && (
+                          <label
+                            for={getToggleAbstractId(paper)}
+                            class="btn-action abstract-toggle"
+                            aria-label="Toggle abstract"
+                            aria-expanded="false"
+                          >
+                            <span class="ml-2 text-sm font-bold text-gray-500 move-button">
+                              ABSTRACT
+                            </span>
+                            <svg
+                              xmlns="http://www.w3.org/2000/svg"
+                              fill="none"
+                              viewBox="0 0 24 24"
+                              class="h-6 w-6 transform transition-transform duration-200"
+                              stroke="currentColor"
+                              stroke-width="2"
+                            >
+                              {/* source https://heroicons.com v1.0.6, MIT licensed */}
+                              <path
+                                stroke-linecap="round"
+                                stroke-linejoin="round"
+                                d="M19 9l-7 7-7-7"
+                              />
+                            </svg>
+                          </label>
+                        )}
+                      </div>
+                    </div>
+                    <div class="abstract-content mt-4 text-sm text-gray-700">
+                      {paper.abstract &&
+                        paper.abstract
+                          .split("\n\n")
+                          .map((paragraph, i) => (
+                            <p class={i > 0 ? "mt-4" : ""}>{paragraph}</p>
+                          ))}
+                    </div>
+                  </article>
+                ))}
+              </div>
+            </section>
+          ))
+        }
+      </div>
+    </Container>
+  </div>
+</Layout>
```

Comment:
Some parts of these templates look really repetitive to me, maybe those could be split up into reusable components. This is currently really hard to read tbh. though it does its job, so good job htere :)

Reply:
In principle I agree with this suggestion.